common/collection/initerface.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package collection
+
+type (
+	// Queue is the interface for queue
+	Queue interface {
+		// PeekItem returns the first item of the queue
+		Peek() interface{}
+		// Offer push an item to the queue
+		Offer(item interface{})
+		// Poll pop an item from the queue
+		Poll() interface{}
+		// IsEmpty indicate if the queue is empty
+		IsEmpty() bool
+		// Size size of the queue
+		Size() int
```

Comment:
i know, just one thing, if we are going to have a base interface for all collection we have, meaning including our concurrent hashmap, then does len make sense for map?

Reply:
sure, will change the function name to add / remove / peek / len

Reply:
Go standard naming convention for this is Len()